src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs

```diff
@@ -121,7 +122,42 @@ private void AddListed()
                     }
                 }
 
-                _metadata["listed"] = listed;
+                _metadata[MetadataConstants.ListedPropertyName] = listed;
+            }
+
+            private void AddSemVerLevelKey()
+            {
+                var version = (string)_catalog[MetadataConstants.VersionPropertyName];
+                if (version != null)
+                {
+                    NuGetVersion packageOriginalVersion;
+                    if (NuGetVersion.TryParse(version, out packageOriginalVersion))
+                    {
+                        if (packageOriginalVersion.IsSemVer2)
+                        {
+                            _metadata[MetadataConstants.SemVerLevelKeyPropertyName] = MetadataConstants.SemVerLevel2Value;
+                            return;
+                        }
+                    }
+                }
+
+                var dependencyGroups = _reader.GetPackageDependencies().ToList();
+                foreach (var dependencyGroup in dependencyGroups)
+                {
+                    if (dependencyGroup.Packages.Any())
```

Comment:
Don't think we need this additional check here, as the enumeration on [line 149](https://github.com/NuGet/NuGet.Services.Metadata/pull/144/files#diff-8e9238f17f25f40d2a5a0ef47ca1901eR149) will just yield no results.